### ISSUE TITLE
BGP PodCIDRs are not advertised after Cilium restart

### ISSUE THREAD

**Issue by NikAleksandrov** [cilium/cilium#24069]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

When I configure BGP peering between a k8s node and remote end with `exportPodCIDR`[1] I get the expected behaviour and the Pod CIDR for the node is properly advertised, but when I do `kubectl -n kube-system rollout restart ds/cilium` and the BGP session is re-established the Pod CIDR is not advertised anymore[3].

Steps:
1. Install cilium 1.13.0:
 `helm install cilium cilium/cilium --version 1.13.0   --namespace kube-system --set bgpControlPlane.enabled=true`
2. Label node1 and configure k8s node BGP as in [1]
3. Configure remote end BGP, bird example in [2]
4. Notice that Pod CIDR is advertised properly as in first part of [3]
5. `kubectl -n kube-system rollout restart ds/cilium`
6. After BGP sessions are re-established notice that the Pod CIDR is missing and there were no updates as in second part of [3]

[1] 192.168.122.1 is the host and remote end, 192.168.122.74 is a k8s node VM, BGP config:
- `kubectl label nodes k8s-node1 name=node1`
- policy
```
apiVersion: "cilium.io/v2alpha1"
kind: CiliumBGPPeeringPolicy
metadata:
  name: bgp-policy-simple
spec:
 nodeSelector:
  matchLabels:
    name: node1
 virtualRouters:
 - localASN: 64555
   exportPodCIDR: true
   neighbors:
    - peerAddress: 192.168.122.1/32
      peerASN: 64555
```

[2] bird config: 192.168.122.1 host, 192.168.122.74 k8s node VM
```
protocol bgp k8s_node {
        description "k8s node";
        hold time 90;
        keepalive time 30;
        local 192.168.122.1 as 64555;
        neighbor 192.168.122.74 as 64555;
        ipv4 {
                import all;
                export none;
        };
}
```

[3] bird logs
- Initially established session with PodCIDR received as can be seen in "Import updates"
```
k8s_node BGP        ---        up     13:06:04.583  Established   
  Description:    k8s node
  BGP state:          Established
    Neighbor address: 192.168.122.74
    Neighbor AS:      64555
    Local AS:         64555
    Neighbor ID:      192.168.122.74
    Local capabilities
      Multiprotocol
        AF announced: ipv4
      Route refresh
      Graceful restart
      4-octet AS numbers
      Enhanced refresh
      Long-lived graceful restart
    Neighbor capabilities
      Multiprotocol
        AF announced: ipv4 ipv6
      Route refresh
      Extended next hop
        IPv6 nexthop: ipv4
      4-octet AS numbers
      Hostname: k8s-node2
    Session:          internal multihop AS4
    Source address:   192.168.122.1
    Hold timer:       86.039/90
    Keepalive timer:  29.136/30
  Channel ipv4
    State:          UP
    Table:          master4
    Preference:     100
    Input filter:   ACCEPT
    Output filter:  REJECT
    Routes:         1 imported, 0 exported, 1 preferred
    Route change stats:     received   rejected   filtered    ignored   accepted
      Import updates:              1          0          0          0          1
      Import withdraws:            0          0        ---          0          0
      Export updates:              5          1          4        ---          0
      Export withdraws:            0        ---        ---        ---          0
    BGP Next hop:   192.168.122.1
    IGP IPv4 table: master4
```

- after Cilium restart and the session is re-established you can see no updates are received
```
k8s_node BGP        ---        up     13:06:41.559  Established   
  Description:    k8s node
  BGP state:          Established
    Neighbor address: 192.168.122.74
    Neighbor AS:      64555
    Local AS:         64555
    Neighbor ID:      192.168.122.74
    Local capabilities
      Multiprotocol
        AF announced: ipv4
      Route refresh
      Graceful restart
      4-octet AS numbers
      Enhanced refresh
      Long-lived graceful restart
    Neighbor capabilities
      Multiprotocol
        AF announced: ipv4 ipv6
      Route refresh
      Extended next hop
        IPv6 nexthop: ipv4
      4-octet AS numbers
      Hostname: k8s-node2
    Session:          internal multihop AS4
    Source address:   192.168.122.1
    Hold timer:       62.280/90
    Keepalive timer:  4.460/30
  Channel ipv4
    State:          UP
    Table:          master4
    Preference:     100
    Input filter:   ACCEPT
    Output filter:  REJECT
    Routes:         0 imported, 0 exported, 0 preferred
    Route change stats:     received   rejected   filtered    ignored   accepted
      Import updates:              0          0          0          0          0
      Import withdraws:            0          0        ---          0          0
      Export updates:              4          0          4        ---          0
      Export withdraws:            0        ---        ---        ---          0
    BGP Next hop:   192.168.122.1
    IGP IPv4 table: master4
```

### Cilium Version

1.13.0

### Kernel Version

5.18.0-rc7+

### Kubernetes Version

v1.26.1

### Relevant log output

```shell
grep -i BGP logs after Cilium restart:

level=info msg="Cilium BGP Control Plane Controller now running..." component=Controller.Run subsys=bgp-control-plane
level=info msg="Cilium BGP Control Plane Controller woken for reconciliation" component=Controller.Run subsys=bgp-control-plane
level=info msg="Registering GoBGP servers for policy with local ASN 64555" component=gobgp.RouterManager.registerBGPServer subsys=bgp-control-plane
level=info msg="Reconciling peers for virtual router with local ASN 64555" component=gobgp.neighborReconciler subsys=bgp-control-plane
level=info msg="Adding peer 192.168.122.1/32 64555 to local ASN 64555" component=gobgp.neighborReconciler subsys=bgp-control-plane
level=info msg="Add a peer configuration" Key=192.168.122.1 Topic=Peer asn=64555 component=gobgp.BgpServerInstance subsys=bgp-control-plane
level=info msg="Done reconciling peers for virtual router with local ASN 64555" component=gobgp.neighborReconciler subsys=bgp-control-plane
level=info msg="Successfully registered GoBGP servers for policy with local ASN 64555" component=gobgp.RouterManager.registerBGPServer subsys=bgp-control-plane
level=info msg="Cilium BGP Control Plane Controller woken for reconciliation" component=Controller.Run subsys=bgp-control-plane
level=info msg="Done reconciling peers for virtual router with local ASN 64555" component=gobgp.neighborReconciler subsys=bgp-control-plane
level=info msg="Cilium BGP Control Plane Controller woken for reconciliation" component=Controller.Run subsys=bgp-control-plane
level=info msg="Virtual router with ASN 64555 router ID has changed from 10.10.2.167 to 192.168.122.74" component=gobgp.preflightReconciler subsys=bgp-control-plane
level=info msg="Recreating virtual router with ASN 64555 for changes to take effect" component=gobgp.preflightReconciler subsys=bgp-control-plane
level=info msg="Delete a peer configuration" Key=192.168.122.1 Topic=Peer asn=64555 component=gobgp.BgpServerInstance subsys=bgp-control-plane
level=info msg="Reconciling peers for virtual router with local ASN 64555" component=gobgp.neighborReconciler subsys=bgp-control-plane
level=info msg="Adding peer 192.168.122.1/32 64555 to local ASN 64555" component=gobgp.neighborReconciler subsys=bgp-control-plane
level=info msg="Add a peer configuration" Key=192.168.122.1 Topic=Peer asn=64555 component=gobgp.BgpServerInstance subsys=bgp-control-plane
level=info msg="Done reconciling peers for virtual router with local ASN 64555" component=gobgp.neighborReconciler subsys=bgp-control-plane
level=info msg="Peer Up" Key=192.168.122.1 State=BGP_FSM_OPENCONFIRM Topic=Peer asn=64555 component=gobgp.BgpServerInstance subsys=bgp-control-plane
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by FilipK-CZ** [comment]
I have the same problem

**Comment by kakkotetsu** [comment]
I have same issue. And in my environment, the way to recover is to recreate(delete -> create) the ciliumbgppeeringpolicy resource.
```
Cilium Version
1.13.0

Kernel Version
5.19.0-35-generic

Kubernetes Version
v1.26.2
```

**Comment by dylandreimerink** [comment]
I made an honest attempt to reproduce the issue, but so far have not been able to.

At first I attempted to repoduce it with our default containerlabs setup which uses FRR. I have the route "10.0.0.0/24", then perform a restart when it goes away, but afterwards it comes back.
```
K>* 0.0.0.0/0 [0/0] via 172.20.20.1, eth0, 00:05:07
B>* 10.0.0.0/24 [20/0] via 172.0.0.2, net0, weight 1, 00:00:07
C>* 172.0.0.0/24 is directly connected, net0, 00:05:07
C>* 172.20.20.0/24 is directly connected, eth0, 00:05:07
tor# show ip route
Codes: K - kernel route, C - connected, S - static, R - RIP,
       O - OSPF, I - IS-IS, B - BGP, E - EIGRP, N - NHRP,
       T - Table, v - VNC, V - VNC-Direct, A - Babel, F - PBR,
       f - OpenFabric,
       > - selected route, * - FIB route, q - queued, r - rejected, b - backup
       t - trapped, o - offload failure

tor# show ip route
Codes: K - kernel route, C - connected, S - static, R - RIP,
       O - OSPF, I - IS-IS, B - BGP, E - EIGRP, N - NHRP,
       T - Table, v - VNC, V - VNC-Direct, A - Babel, F - PBR,
       f - OpenFabric,
       > - selected route, * - FIB route, q - queued, r - rejected, b - backup
       t - trapped, o - offload failure

K>* 0.0.0.0/0 [0/0] via 172.20.20.1, eth0, 00:05:30
C>* 172.0.0.0/24 is directly connected, net0, 00:05:30
C>* 172.20.20.0/24 is directly connected, eth0, 00:05:30
tor# show ip route
Codes: K - kernel route, C - connected, S - static, R - RIP,
       O - OSPF, I - IS-IS, B - BGP, E - EIGRP, N - NHRP,
       T - Table, v - VNC, V - VNC-Direct, A - Babel, F - PBR,
       f - OpenFabric,
       > - selected route, * - FIB route, q - queued, r - rejected, b - backup
       t - trapped, o - offload failure

K>* 0.0.0.0/0 [0/0] via 172.20.20.1, eth0, 00:05:34
B>* 10.0.0.0/24 [20/0] via 172.0.0.2, net0, weight 1, 00:00:03
C>* 172.0.0.0/24 is directly connected, net0, 00:05:34
C>* 172.20.20.0/24 is directly connected, eth0, 00:05:34
```

I modified the containerlabs environment to add a bird router to match the repro steps exactly:
```
name: bgp-cplane-devel
topology:
  kinds:
    linux:
      cmd: bash
  nodes:
    tor:
      kind: linux
      image: frrouting/frr:v8.2.2
      exec:
        # peer over this link with cilium
        - ip addr add 172.0.0.1/24 dev net0
        # Boiler plate to make FRR work
        - touch /etc/frr/vtysh.conf
        - sed -i -e 's/bgpd=no/bgpd=yes/g' /etc/frr/daemons
        - /usr/lib/frr/frrinit.sh start
        - >-
          vtysh -c 'conf t'
          -c 'debug bgp neighbor-events'
          -c 'debug bgp updates'
          -c 'debug bgp zebra'
          -c '! peer with Cilium'
          -c 'router bgp 65000'
          -c ' no bgp ebgp-requires-policy'    
          -c ' bgp router-id 172.0.0.1'
          -c ' neighbor 172.0.0.2 remote-as 65001'
          -c ' address-family ipv6 unicast'
          -c '  neighbor 172.0.0.2  activate'
          -c ' exit-address-family'
          -c '!'
    cilium:
      kind: linux
      image: nicolaka/netshoot:latest
      network-mode: container:control-plane
      exec:
        # peer over this link with tor
      - ip addr add 172.0.0.2/24 dev net0
      - ip addr add 173.0.0.2/24 dev net1
    bird:
      kind: linux
      image: pierky/bird:2.0.11
      binds:
        - bird/bird.conf:/etc/bird/bird.conf
      exec:
        # peer over this link with cilium
        - ip addr add 173.0.0.1/24 dev net0
        # Boiler plate to make bird work
        - bird -c /etc/bird/bird.conf
  links:
    - endpoints: ["tor:net0", "cilium:net0"]
    - endpoints: ["bird:net0", "cilium:net1"]
```

Only altering IPs.

```
protocol bgp k8s_node {
    description "k8s node";
    hold time 90;
    keepalive time 30;
    local 173.0.0.1 as 64555;
    neighbor 173.0.0.2 as 64555;
    ipv4 {
    import all;
    export none;
    };
}
```

But here the route also comes back after the restart.

```
bird> show route
Table master4:
10.0.0.0/24          unreachable [k8s_node 13:52:02.064 from 173.0.0.2] * (100) [i]
bird> show protocols
Name       Proto      Table      State  Since         Info
k8s_node   BGP        ---        start  13:52:40.797  Active        Socket: Connection refused
bird> show protocols
Name       Proto      Table      State  Since         Info
k8s_node   BGP        ---        start  13:52:40.797  Active        Socket: Connection refused
bird> show protocols
Name       Proto      Table      State  Since         Info
k8s_node   BGP        ---        start  13:52:40.797  Active        Socket: Connection refused
bird> show protocols
Name       Proto      Table      State  Since         Info
k8s_node   BGP        ---        up     13:52:55.429  Established   
bird> show route
Table master4:
10.0.0.0/24          unreachable [k8s_node 13:52:55.429 from 173.0.0.2] * (100) [i]
```

```
level=debug msg="Executing start hook" function="*resource.resource[*github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1.CiliumBGPPeeringPolicy].Start" subsys=hive
level=info msg="Start hook executed" duration="1.079µs" function="*resource.resource[*github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1.CiliumBGPPeeringPolicy].Start" subsys=hive
level=info msg="Cilium BGP Control Plane Controller now running..." component=Controller.Run subsys=bgp-control-plane
level=info msg="Cilium BGP Control Plane Controller woken for reconciliation" component=Controller.Run subsys=bgp-control-plane
level=debug msg="Successfully listed CiliumBGPPeeringPolicies" component=Controller.Reconcile count=1 subsys=bgp-control-plane
level=debug msg="Comparing BGP policy node selector with node's labels" component=PolicySelection nodeLabels="beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=clab-bgp-cplane-devel-control-plane,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node-role.kubernetes.io/master=,node.kubernetes.io/exclude-from-external-load-balancers=" policyNodeSelector="kubernetes.io/hostname=clab-bgp-cplane-devel-control-plane" subsys=bgp-control-plane
level=debug msg="Asking configured BGPRouterManager to configure peering" component=Controller.Reconcile subsys=bgp-control-plane
level=debug msg="Reconciling new CiliumBGPPeeringPolicy" component=gobgp.RouterManager.ConfigurePeers diff="Registering: [64555] Withdrawing: [] Reconciling: []" subsys=bgp-control-plane
level=info msg="Registering GoBGP servers for policy with local ASN 64555" component=gobgp.RouterManager.registerBGPServer subsys=bgp-control-plane
level=debug msg="state changed" Key=173.0.0.1 Topic=Peer asn=64555 component=gobgp.BgpServerInstance new=BGP_FSM_ACTIVE old=BGP_FSM_IDLE reason=idle-hold-timer-expired subsys=bgp-control-plane
level=info msg="Successfully registered GoBGP servers for policy with local ASN 64555" component=gobgp.RouterManager.registerBGPServer subsys=bgp-control-plane
level=info msg="Cilium BGP Control Plane Controller woken for reconciliation" component=Controller.Run subsys=bgp-control-plane
level=debug msg="Successfully listed CiliumBGPPeeringPolicies" component=Controller.Reconcile count=1 subsys=bgp-control-plane
level=debug msg="Comparing BGP policy node selector with node's labels" component=PolicySelection nodeLabels="beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=clab-bgp-cplane-devel-control-plane,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node-role.kubernetes.io/master=,node.kubernetes.io/exclude-from-external-load-balancers=" policyNodeSelector="kubernetes.io/hostname=clab-bgp-cplane-devel-control-plane" subsys=bgp-control-plane
level=debug msg="Asking configured BGPRouterManager to configure peering" component=Controller.Reconcile subsys=bgp-control-plane
level=debug msg="Reconciling new CiliumBGPPeeringPolicy" component=gobgp.RouterManager.ConfigurePeers diff="Registering: [] Withdrawing: [] Reconciling: [64555]" subsys=bgp-control-plane
level=info msg="Cilium BGP Control Plane Controller woken for reconciliation" component=Controller.Run subsys=bgp-control-plane
level=debug msg="Successfully listed CiliumBGPPeeringPolicies" component=Controller.Reconcile count=1 subsys=bgp-control-plane
level=debug msg="Comparing BGP policy node selector with node's labels" component=PolicySelection nodeLabels="beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=clab-bgp-cplane-devel-control-plane,kubernetes.io/os=linux,node-role.kubernetes.io/control-plane=,node-role.kubernetes.io/master=,node.kubernetes.io/exclude-from-external-load-balancers=" policyNodeSelector="kubernetes.io/hostname=clab-bgp-cplane-devel-control-plane" subsys=bgp-control-plane
level=debug msg="Asking configured BGPRouterManager to configure peering" component=Controller.Reconcile subsys=bgp-control-plane
level=debug msg="Reconciling new CiliumBGPPeeringPolicy" component=gobgp.RouterManager.ConfigurePeers diff="Registering: [] Withdrawing: [] Reconciling: [64555]" subsys=bgp-control-plane
level=debug msg="state changed" Key=173.0.0.1 Topic=Peer asn=64555 component=gobgp.BgpServerInstance new=BGP_FSM_OPENSENT old=BGP_FSM_ACTIVE reason=new-connection subsys=bgp-control-plane
level=debug msg="state changed" Key=173.0.0.1 Topic=Peer asn=64555 component=gobgp.BgpServerInstance new=BGP_FSM_OPENCONFIRM old=BGP_FSM_OPENSENT reason=open-msg-received subsys=bgp-control-plane
level=info msg="Peer Up" Key=173.0.0.1 State=BGP_FSM_OPENCONFIRM Topic=Peer asn=64555 component=gobgp.BgpServerInstance subsys=bgp-control-plane
level=debug msg="state changed" Key=173.0.0.1 Topic=Peer asn=64555 component=gobgp.BgpServerInstance new=BGP_FSM_ESTABLISHED old=BGP_FSM_OPENCONFIRM reason=open-msg-negotiated subsys=bgp-control-plane
level=debug msg="sent update" Key=173.0.0.1 State=BGP_FSM_ESTABLISHED Topic=Peer asn=64555 attributes="[{Origin: i}  {Nexthop: 173.0.0.2} {LocalPref: 100}]" component=gobgp.BgpServerInstance nlri="[10.0.0.0/24]" subsys=bgp-control-plane withdrawals="[]"
level=debug msg=sent Key=173.0.0.1 State=BGP_FSM_ESTABLISHED Topic=Peer asn=64555 component=gobgp.BgpServerInstance data="&{{[] 19 4} 0x5b3cce8}" subsys=bgp-control-plane
level=debug msg=sent Key=173.0.0.1 State=BGP_FSM_ESTABLISHED Topic=Peer asn=64555 component=gobgp.BgpServerInstance data="&{{[] 19 4} 0x5b3cce8}" subsys=bgp-control-plane
level=debug msg=sent Key=173.0.0.1 State=BGP_FSM_ESTABLISHED Topic=Peer asn=64555 component=gobgp.BgpServerInstance data="&{{[] 19 4} 0x5b3cce8}" subsys=bgp-control-plane
level=debug msg=sent Key=173.0.0.1 State=BGP_FSM_ESTABLISHED Topic=Peer asn=64555 component=gobgp.BgpServerInstance data="&{{[] 19 4} 0x5b3cce8}" subsys=bgp-control-plane
level=debug msg=sent Key=173.0.0.1 State=BGP_FSM_ESTABLISHED Topic=Peer asn=64555 component=gobgp.BgpServerInstance data="&{{[] 19 4} 0x5b3cce8}" subsys=bgp-control-plane
level=debug msg=sent Key=173.0.0.1 State=BGP_FSM_ESTABLISHED Topic=Peer asn=64555 component=gobgp.BgpServerInstance data="&{{[] 19 4} 0x5b3cce8}" subsys=bgp-control-plane
level=debug msg=sent Key=173.0.0.1 State=BGP_FSM_ESTABLISHED Topic=Peer asn=64555 component=gobgp.BgpServerInstance data="&{{[] 19 4} 0x5b3cce8}" subsys=bgp-control-plane
level=debug msg=sent Key=173.0.0.1 State=BGP_FSM_ESTABLISHED Topic=Peer asn=64555 component=gobgp.BgpServerInstance data="&{{[] 19 4} 0x5b3cce8}" subsys=bgp-control-plane
level=debug msg=sent Key=173.0.0.1 State=BGP_FSM_ESTABLISHED Topic=Peer asn=64555 component=gobgp.BgpServerInstance data="&{{[] 19 4} 0x5b3cce8}" subsys=bgp-control-plane
level=debug msg=sent Key=173.0.0.1 State=BGP_FSM_ESTABLISHED Topic=Peer asn=64555 component=gobgp.BgpServerInstance data="&{{[] 19 4} 0x5b3cce8}" subsys=bgp-control-plane
level=debug msg=sent Key=173.0.0.1 State=BGP_FSM_ESTABLISHED Topic=Peer asn=64555 component=gobgp.BgpServerInstance data="&{{[] 19 4} 0x5b3cce8}" subsys=bgp-control-plane
level=debug msg=sent Key=173.0.0.1 State=BGP_FSM_ESTABLISHED Topic=Peer asn=64555 component=gobgp.BgpServerInstance data="&{{[] 19 4} 0x5b3cce8}" subsys=bgp-control-plane
level=debug msg=sent Key=173.0.0.1 State=BGP_FSM_ESTABLISHED Topic=Peer asn=64555 component=gobgp.BgpServerInstance data="&{{[] 19 4} 0x5b3cce8}" subsys=bgp-control-plane
level=debug msg=sent Key=173.0.0.1 State=BGP_FSM_ESTABLISHED Topic=Peer asn=64555 component=gobgp.BgpServerInstance data="&{{[] 19 4} 0x5b3cce8}" subsys=bgp-control-plane
```

So there is a factor required for reproduction that we are missing here. @NikAleksandrov do you have any ideas about what that could be?

**Comment by NikAleksandrov** [comment]
@dylandreimerink not sure what's happening with your setup, I just quickly reproduced it using a kind cluster, here are the exact steps:
1. create kind cluster with the following config:
```
kind: Cluster
apiVersion: kind.x-k8s.io/v1alpha4
nodes:
- role: control-plane
- role: worker
- role: worker
networking:
  disableDefaultCNI: true
```

`$ kind create cluster --name test-cluster --config ./kind-cluster.yaml`

2. install cilium
`$ helm install cilium cilium/cilium --version 1.13.0   --namespace kube-system --set bgpControlPlane.enabled=true`

3. label first worker
`$ kubectl label nodes test-cluster-worker name=node1`

4. apply BGP config (192.168.122.1 is on the host of the kind cluster)
```
apiVersion: "cilium.io/v2alpha1"
kind: CiliumBGPPeeringPolicy
metadata:
  name: bgp-policy-simple
spec:
 nodeSelector:
  matchLabels:
    name: node1
 virtualRouters:
 - localASN: 64555
   exportPodCIDR: true
   neighbors:
    - peerAddress: 192.168.122.1/32
      peerASN: 64555
```

`$ kubectl apply -f ./bgp.pol`

5. on the host side (bird running, config is below) notice the route is advertised:
```
k8s_node BGP        ---        up     17:02:29.836  Established   
  Description:    k8s node
  BGP state:          Established
    Neighbor address: 172.18.0.4
    Neighbor AS:      64555
    Local AS:         64555
    Neighbor ID:      172.18.0.4
    Local capabilities
      Multiprotocol
        AF announced: ipv4
      Route refresh
      Graceful restart
      4-octet AS numbers
      Enhanced refresh
      Long-lived graceful restart
    Neighbor capabilities
      Multiprotocol
        AF announced: ipv4 ipv6
      Route refresh
      Extended next hop
        IPv6 nexthop: ipv4
      4-octet AS numbers
      Hostname: test-cluster-worker
    Session:          internal multihop AS4
    Source address:   192.168.122.1
    Hold timer:       71.648/90
    Keepalive timer:  6.219/30
  Channel ipv4
    State:          UP
    Table:          master4
    Preference:     100
    Input filter:   ACCEPT
    Output filter:  REJECT
    Routes:         1 imported, 0 exported, 1 preferred
    Route change stats:     received   rejected   filtered    ignored   accepted
      Import updates:              1          0          0          0          1
      Import withdraws:            0          0        ---          0          0
      Export updates:              7          1          6        ---          0
      Export withdraws:            0        ---        ---        ---          0
    BGP Next hop:   192.168.122.1
    IGP IPv4 table: master4
```

5. restart cilium:
`$ kubectl -n kube-system rollout restart ds/cilium`

6. wait for the BGP session to be re-established and print it again, notice no import updates:
```
k8s_node BGP        ---        up     17:08:35.425  Established   
  Description:    k8s node
  BGP state:          Established
    Neighbor address: 172.18.0.4
    Neighbor AS:      64555
    Local AS:         64555
    Neighbor ID:      172.18.0.4
    Local capabilities
      Multiprotocol
        AF announced: ipv4
      Route refresh
      Graceful restart
      4-octet AS numbers
      Enhanced refresh
      Long-lived graceful restart
    Neighbor capabilities
      Multiprotocol
        AF announced: ipv4 ipv6
      Route refresh
      Extended next hop
        IPv6 nexthop: ipv4
      4-octet AS numbers
      Hostname: test-cluster-worker
    Session:          internal multihop AS4
    Source address:   192.168.122.1
    Hold timer:       80.547/90
    Keepalive timer:  24.964/30
  Channel ipv4
    State:          UP
    Table:          master4
    Preference:     100
    Input filter:   ACCEPT
    Output filter:  REJECT
    Routes:         0 imported, 0 exported, 0 preferred
    Route change stats:     received   rejected   filtered    ignored   accepted
      Import updates:              0          0          0          0          0
      Import withdraws:            0          0        ---          0          0
      Export updates:              6          0          6        ---          0
      Export withdraws:            0        ---        ---        ---          0
    BGP Next hop:   192.168.122.1
    IGP IPv4 table: master4
```

Logs from Cilium running on the kind worker node after the restart:
```
$ kubectl -n kube-system logs cilium-2vml9
Defaulted container "cilium-agent" out of: cilium-agent, config (init), mount-cgroup (init), apply-sysctl-overwrites (init), mount-bpf-fs (init), clean-cilium-state (init)
level=info msg="Memory available for map entries (0.003% of 67340828672B): 168352071B" subsys=config
level=info msg="option bpf-ct-global-tcp-max set by dynamic sizing to 590709" subsys=config
level=info msg="option bpf-ct-global-any-max set by dynamic sizing to 295354" subsys=config
level=info msg="option bpf-nat-global-max set by dynamic sizing to 590709" subsys=config
level=info msg="option bpf-neigh-global-max set by dynamic sizing to 590709" subsys=config
level=info msg="option bpf-sock-rev-map-max set by dynamic sizing to 295354" subsys=config
level=info msg="  --agent-health-port='9879'" subsys=daemon
level=info msg="  --agent-labels=''" subsys=daemon
level=info msg="  --agent-not-ready-taint-key='node.cilium.io/agent-not-ready'" subsys=daemon
level=info msg="  --allocator-list-timeout='3m0s'" subsys=daemon
level=info msg="  --allow-icmp-frag-needed='true'" subsys=daemon
level=info msg="  --allow-localhost='auto'" subsys=daemon
level=info msg="  --annotate-k8s-node='false'" subsys=daemon
level=info msg="  --api-rate-limit=''" subsys=daemon
level=info msg="  --arping-refresh-period='30s'" subsys=daemon
level=info msg="  --auto-create-cilium-node-resource='true'" subsys=daemon
level=info msg="  --auto-direct-node-routes='false'" subsys=daemon
level=info msg="  --bgp-announce-lb-ip='false'" subsys=daemon
level=info msg="  --bgp-announce-pod-cidr='false'" subsys=daemon
level=info msg="  --bgp-config-path='/var/lib/cilium/bgp/config.yaml'" subsys=daemon
level=info msg="  --bpf-ct-global-any-max='262144'" subsys=daemon
level=info msg="  --bpf-ct-global-tcp-max='524288'" subsys=daemon
level=info msg="  --bpf-ct-timeout-regular-any='1m0s'" subsys=daemon
level=info msg="  --bpf-ct-timeout-regular-tcp='6h0m0s'" subsys=daemon
level=info msg="  --bpf-ct-timeout-regular-tcp-fin='10s'" subsys=daemon
level=info msg="  --bpf-ct-timeout-regular-tcp-syn='1m0s'" subsys=daemon
level=info msg="  --bpf-ct-timeout-service-any='1m0s'" subsys=daemon
level=info msg="  --bpf-ct-timeout-service-tcp='6h0m0s'" subsys=daemon
level=info msg="  --bpf-ct-timeout-service-tcp-grace='1m0s'" subsys=daemon
level=info msg="  --bpf-filter-priority='1'" subsys=daemon
level=info msg="  --bpf-fragments-map-max='8192'" subsys=daemon
level=info msg="  --bpf-lb-acceleration='disabled'" subsys=daemon
level=info msg="  --bpf-lb-affinity-map-max='0'" subsys=daemon
level=info msg="  --bpf-lb-algorithm='random'" subsys=daemon
level=info msg="  --bpf-lb-dev-ip-addr-inherit=''" subsys=daemon
level=info msg="  --bpf-lb-dsr-dispatch='opt'" subsys=daemon
level=info msg="  --bpf-lb-dsr-l4-xlate='frontend'" subsys=daemon
level=info msg="  --bpf-lb-external-clusterip='false'" subsys=daemon
level=info msg="  --bpf-lb-maglev-hash-seed='JLfvgnHc2kaSUFaI'" subsys=daemon
level=info msg="  --bpf-lb-maglev-map-max='0'" subsys=daemon
level=info msg="  --bpf-lb-maglev-table-size='16381'" subsys=daemon
level=info msg="  --bpf-lb-map-max='65536'" subsys=daemon
level=info msg="  --bpf-lb-mode='snat'" subsys=daemon
level=info msg="  --bpf-lb-rev-nat-map-max='0'" subsys=daemon
level=info msg="  --bpf-lb-rss-ipv4-src-cidr=''" subsys=daemon
level=info msg="  --bpf-lb-rss-ipv6-src-cidr=''" subsys=daemon
level=info msg="  --bpf-lb-service-backend-map-max='0'" subsys=daemon
level=info msg="  --bpf-lb-service-map-max='0'" subsys=daemon
level=info msg="  --bpf-lb-sock='false'" subsys=daemon
level=info msg="  --bpf-lb-sock-hostns-only='false'" subsys=daemon
level=info msg="  --bpf-lb-source-range-map-max='0'" subsys=daemon
level=info msg="  --bpf-map-dynamic-size-ratio='0.0025'" subsys=daemon
level=info msg="  --bpf-map-event-buffers=''" subsys=daemon
level=info msg="  --bpf-nat-global-max='524288'" subsys=daemon
level=info msg="  --bpf-neigh-global-max='524288'" subsys=daemon
level=info msg="  --bpf-policy-map-max='16384'" subsys=daemon
level=info msg="  --bpf-root='/sys/fs/bpf'" subsys=daemon
level=info msg="  --bpf-sock-rev-map-max='262144'" subsys=daemon
level=info msg="  --bypass-ip-availability-upon-restore='false'" subsys=daemon
level=info msg="  --certificates-directory='/var/run/cilium/certs'" subsys=daemon
level=info msg="  --cflags=''" subsys=daemon
level=info msg="  --cgroup-root='/run/cilium/cgroupv2'" subsys=daemon
level=info msg="  --cilium-endpoint-gc-interval='5m0s'" subsys=daemon
level=info msg="  --cluster-health-port='4240'" subsys=daemon
level=info msg="  --cluster-id='0'" subsys=daemon
level=info msg="  --cluster-name='default'" subsys=daemon
level=info msg="  --cluster-pool-ipv4-cidr='10.0.0.0/8'" subsys=daemon
level=info msg="  --cluster-pool-ipv4-mask-size='24'" subsys=daemon
level=info msg="  --clustermesh-config='/var/lib/cilium/clustermesh/'" subsys=daemon
level=info msg="  --cmdref=''" subsys=daemon
level=info msg="  --cni-chaining-mode=''" subsys=daemon
level=info msg="  --config=''" subsys=daemon
level=info msg="  --config-dir='/tmp/cilium/config-map'" subsys=daemon
level=info msg="  --config-sources='config-map:kube-system/cilium-config'" subsys=daemon
level=info msg="  --conntrack-gc-interval='0s'" subsys=daemon
level=info msg="  --crd-wait-timeout='5m0s'" subsys=daemon
level=info msg="  --custom-cni-conf='false'" subsys=daemon
level=info msg="  --datapath-mode='veth'" subsys=daemon
level=info msg="  --debug='false'" subsys=daemon
level=info msg="  --debug-verbose=''" subsys=daemon
level=info msg="  --derive-masquerade-ip-addr-from-device=''" subsys=daemon
level=info msg="  --devices=''" subsys=daemon
level=info msg="  --direct-routing-device=''" subsys=daemon
level=info msg="  --disable-cnp-status-updates='true'" subsys=daemon
level=info msg="  --disable-endpoint-crd='false'" subsys=daemon
level=info msg="  --disable-envoy-version-check='false'" subsys=daemon
level=info msg="  --disable-iptables-feeder-rules=''" subsys=daemon
level=info msg="  --dns-max-ips-per-restored-rule='1000'" subsys=daemon
level=info msg="  --dns-policy-unload-on-shutdown='false'" subsys=daemon
level=info msg="  --dnsproxy-concurrency-limit='0'" subsys=daemon
level=info msg="  --dnsproxy-concurrency-processing-grace-period='0s'" subsys=daemon
level=info msg="  --dnsproxy-lock-count='128'" subsys=daemon
level=info msg="  --dnsproxy-lock-timeout='500ms'" subsys=daemon
level=info msg="  --egress-masquerade-interfaces=''" subsys=daemon
level=info msg="  --egress-multi-home-ip-rule-compat='false'" subsys=daemon
level=info msg="  --enable-auto-protect-node-port-range='true'" subsys=daemon
level=info msg="  --enable-bandwidth-manager='false'" subsys=daemon
level=info msg="  --enable-bbr='false'" subsys=daemon
level=info msg="  --enable-bgp-control-plane='true'" subsys=daemon
level=info msg="  --enable-bpf-clock-probe='true'" subsys=daemon
level=info msg="  --enable-bpf-masquerade='false'" subsys=daemon
level=info msg="  --enable-bpf-tproxy='false'" subsys=daemon
level=info msg="  --enable-cilium-endpoint-slice='false'" subsys=daemon
level=info msg="  --enable-custom-calls='false'" subsys=daemon
level=info msg="  --enable-endpoint-health-checking='true'" subsys=daemon
level=info msg="  --enable-endpoint-routes='false'" subsys=daemon
level=info msg="  --enable-envoy-config='false'" subsys=daemon
level=info msg="  --enable-external-ips='true'" subsys=daemon
level=info msg="  --enable-health-check-nodeport='true'" subsys=daemon
level=info msg="  --enable-health-checking='true'" subsys=daemon
level=info msg="  --enable-host-firewall='false'" subsys=daemon
level=info msg="  --enable-host-legacy-routing='false'" subsys=daemon
level=info msg="  --enable-host-port='true'" subsys=daemon
level=info msg="  --enable-hubble='true'" subsys=daemon
level=info msg="  --enable-hubble-recorder-api='true'" subsys=daemon
level=info msg="  --enable-icmp-rules='true'" subsys=daemon
level=info msg="  --enable-identity-mark='true'" subsys=daemon
level=info msg="  --enable-ip-masq-agent='false'" subsys=daemon
level=info msg="  --enable-ipsec='false'" subsys=daemon
level=info msg="  --enable-ipv4='true'" subsys=daemon
level=info msg="  --enable-ipv4-egress-gateway='false'" subsys=daemon
level=info msg="  --enable-ipv4-fragment-tracking='true'" subsys=daemon
level=info msg="  --enable-ipv4-masquerade='true'" subsys=daemon
level=info msg="  --enable-ipv6='false'" subsys=daemon
level=info msg="  --enable-ipv6-big-tcp='false'" subsys=daemon
level=info msg="  --enable-ipv6-masquerade='true'" subsys=daemon
level=info msg="  --enable-ipv6-ndp='false'" subsys=daemon
level=info msg="  --enable-k8s-api-discovery='false'" subsys=daemon
level=info msg="  --enable-k8s-endpoint-slice='true'" subsys=daemon
level=info msg="  --enable-k8s-event-handover='false'" subsys=daemon
level=info msg="  --enable-k8s-terminating-endpoint='true'" subsys=daemon
level=info msg="  --enable-l2-neigh-discovery='true'" subsys=daemon
level=info msg="  --enable-l7-proxy='true'" subsys=daemon
level=info msg="  --enable-local-node-route='true'" subsys=daemon
level=info msg="  --enable-local-redirect-policy='false'" subsys=daemon
level=info msg="  --enable-mke='false'" subsys=daemon
level=info msg="  --enable-monitor='true'" subsys=daemon
level=info msg="  --enable-nat46x64-gateway='false'" subsys=daemon
level=info msg="  --enable-node-port='false'" subsys=daemon
level=info msg="  --enable-pmtu-discovery='false'" subsys=daemon
level=info msg="  --enable-policy='default'" subsys=daemon
level=info msg="  --enable-recorder='false'" subsys=daemon
level=info msg="  --enable-remote-node-identity='true'" subsys=daemon
level=info msg="  --enable-runtime-device-detection='false'" subsys=daemon
level=info msg="  --enable-sctp='false'" subsys=daemon
level=info msg="  --enable-service-topology='false'" subsys=daemon
level=info msg="  --enable-session-affinity='false'" subsys=daemon
level=info msg="  --enable-srv6='false'" subsys=daemon
level=info msg="  --enable-stale-cilium-endpoint-cleanup='true'" subsys=daemon
level=info msg="  --enable-svc-source-range-check='true'" subsys=daemon
level=info msg="  --enable-tracing='false'" subsys=daemon
level=info msg="  --enable-unreachable-routes='false'" subsys=daemon
level=info msg="  --enable-vtep='false'" subsys=daemon
level=info msg="  --enable-well-known-identities='false'" subsys=daemon
level=info msg="  --enable-wireguard='false'" subsys=daemon
level=info msg="  --enable-wireguard-userspace-fallback='false'" subsys=daemon
level=info msg="  --enable-xdp-prefilter='false'" subsys=daemon
level=info msg="  --enable-xt-socket-fallback='true'" subsys=daemon
level=info msg="  --encrypt-interface=''" subsys=daemon
level=info msg="  --encrypt-node='false'" subsys=daemon
level=info msg="  --endpoint-gc-interval='5m0s'" subsys=daemon
level=info msg="  --endpoint-queue-size='25'" subsys=daemon
level=info msg="  --endpoint-status=''" subsys=daemon
level=info msg="  --envoy-config-timeout='2m0s'" subsys=daemon
level=info msg="  --envoy-log=''" subsys=daemon
level=info msg="  --exclude-local-address=''" subsys=daemon
level=info msg="  --fixed-identity-mapping=''" subsys=daemon
level=info msg="  --force-local-policy-eval-at-source='false'" subsys=daemon
level=info msg="  --fqdn-regex-compile-lru-size='1024'" subsys=daemon
level=info msg="  --gops-port='9890'" subsys=daemon
level=info msg="  --http-403-msg=''" subsys=daemon
level=info msg="  --http-idle-timeout='0'" subsys=daemon
level=info msg="  --http-max-grpc-timeout='0'" subsys=daemon
level=info msg="  --http-normalize-path='true'" subsys=daemon
level=info msg="  --http-request-timeout='3600'" subsys=daemon
level=info msg="  --http-retry-count='3'" subsys=daemon
level=info msg="  --http-retry-timeout='0'" subsys=daemon
level=info msg="  --hubble-disable-tls='false'" subsys=daemon
level=info msg="  --hubble-event-buffer-capacity='4095'" subsys=daemon
level=info msg="  --hubble-event-queue-size='0'" subsys=daemon
level=info msg="  --hubble-export-file-compress='false'" subsys=daemon
level=info msg="  --hubble-export-file-max-backups='5'" subsys=daemon
level=info msg="  --hubble-export-file-max-size-mb='10'" subsys=daemon
level=info msg="  --hubble-export-file-path=''" subsys=daemon
level=info msg="  --hubble-listen-address=':4244'" subsys=daemon
level=info msg="  --hubble-metrics=''" subsys=daemon
level=info msg="  --hubble-metrics-server=''" subsys=daemon
level=info msg="  --hubble-prefer-ipv6='false'" subsys=daemon
level=info msg="  --hubble-recorder-sink-queue-size='1024'" subsys=daemon
level=info msg="  --hubble-recorder-storage-path='/var/run/cilium/pcaps'" subsys=daemon
level=info msg="  --hubble-skip-unknown-cgroup-ids='true'" subsys=daemon
level=info msg="  --hubble-socket-path='/var/run/cilium/hubble.sock'" subsys=daemon
level=info msg="  --hubble-tls-cert-file='/var/lib/cilium/tls/hubble/server.crt'" subsys=daemon
level=info msg="  --hubble-tls-client-ca-files='/var/lib/cilium/tls/hubble/client-ca.crt'" subsys=daemon
level=info msg="  --hubble-tls-key-file='/var/lib/cilium/tls/hubble/server.key'" subsys=daemon
level=info msg="  --identity-allocation-mode='crd'" subsys=daemon
level=info msg="  --identity-change-grace-period='5s'" subsys=daemon
level=info msg="  --identity-gc-interval='15m0s'" subsys=daemon
level=info msg="  --identity-heartbeat-timeout='30m0s'" subsys=daemon
level=info msg="  --identity-restore-grace-period='10m0s'" subsys=daemon
level=info msg="  --install-egress-gateway-routes='false'" subsys=daemon
level=info msg="  --install-iptables-rules='true'" subsys=daemon
level=info msg="  --install-no-conntrack-iptables-rules='false'" subsys=daemon
level=info msg="  --ip-allocation-timeout='2m0s'" subsys=daemon
level=info msg="  --ip-masq-agent-config-path='/etc/config/ip-masq-agent'" subsys=daemon
level=info msg="  --ipam='cluster-pool'" subsys=daemon
level=info msg="  --ipsec-key-file=''" subsys=daemon
level=info msg="  --iptables-lock-timeout='5s'" subsys=daemon
level=info msg="  --iptables-random-fully='false'" subsys=daemon
level=info msg="  --ipv4-native-routing-cidr=''" subsys=daemon
level=info msg="  --ipv4-node='auto'" subsys=daemon
level=info msg="  --ipv4-pod-subnets=''" subsys=daemon
level=info msg="  --ipv4-range='auto'" subsys=daemon
level=info msg="  --ipv4-service-loopback-address='169.254.42.1'" subsys=daemon
level=info msg="  --ipv4-service-range='auto'" subsys=daemon
level=info msg="  --ipv6-cluster-alloc-cidr='f00d::/64'" subsys=daemon
level=info msg="  --ipv6-mcast-device=''" subsys=daemon
level=info msg="  --ipv6-native-routing-cidr=''" subsys=daemon
level=info msg="  --ipv6-node='auto'" subsys=daemon
level=info msg="  --ipv6-pod-subnets=''" subsys=daemon
level=info msg="  --ipv6-range='auto'" subsys=daemon
level=info msg="  --ipv6-service-range='auto'" subsys=daemon
level=info msg="  --join-cluster='false'" subsys=daemon
level=info msg="  --k8s-api-server=''" subsys=daemon
level=info msg="  --k8s-client-burst='0'" subsys=daemon
level=info msg="  --k8s-client-qps='0'" subsys=daemon
level=info msg="  --k8s-heartbeat-timeout='30s'" subsys=daemon
level=info msg="  --k8s-kubeconfig-path=''" subsys=daemon
level=info msg="  --k8s-namespace='kube-system'" subsys=daemon
level=info msg="  --k8s-require-ipv4-pod-cidr='false'" subsys=daemon
level=info msg="  --k8s-require-ipv6-pod-cidr='false'" subsys=daemon
level=info msg="  --k8s-service-cache-size='128'" subsys=daemon
level=info msg="  --k8s-service-proxy-name=''" subsys=daemon
level=info msg="  --k8s-sync-timeout='3m0s'" subsys=daemon
level=info msg="  --k8s-watcher-endpoint-selector='metadata.name!=kube-scheduler,metadata.name!=kube-controller-manager,metadata.name!=etcd-operator,metadata.name!=gcp-controller-manager'" subsys=daemon
level=info msg="  --keep-config='false'" subsys=daemon
level=info msg="  --kube-proxy-replacement='disabled'" subsys=daemon
level=info msg="  --kube-proxy-replacement-healthz-bind-address=''" subsys=daemon
level=info msg="  --kvstore=''" subsys=daemon
level=info msg="  --kvstore-connectivity-timeout='2m0s'" subsys=daemon
level=info msg="  --kvstore-lease-ttl='15m0s'" subsys=daemon
level=info msg="  --kvstore-max-consecutive-quorum-errors='2'" subsys=daemon
level=info msg="  --kvstore-opt=''" subsys=daemon
level=info msg="  --kvstore-periodic-sync='5m0s'" subsys=daemon
level=info msg="  --label-prefix-file=''" subsys=daemon
level=info msg="  --labels=''" subsys=daemon
level=info msg="  --lib-dir='/var/lib/cilium'" subsys=daemon
level=info msg="  --local-max-addr-scope='252'" subsys=daemon
level=info msg="  --local-router-ipv4=''" subsys=daemon
level=info msg="  --local-router-ipv6=''" subsys=daemon
level=info msg="  --log-driver=''" subsys=daemon
level=info msg="  --log-opt=''" subsys=daemon
level=info msg="  --log-system-load='false'" subsys=daemon
level=info msg="  --max-controller-interval='0'" subsys=daemon
level=info msg="  --metrics=''" subsys=daemon
level=info msg="  --mke-cgroup-mount=''" subsys=daemon
level=info msg="  --monitor-aggregation='medium'" subsys=daemon
level=info msg="  --monitor-aggregation-flags='all'" subsys=daemon
level=info msg="  --monitor-aggregation-interval='5s'" subsys=daemon
level=info msg="  --monitor-queue-size='0'" subsys=daemon
level=info msg="  --mtu='0'" subsys=daemon
level=info msg="  --node-port-acceleration='disabled'" subsys=daemon
level=info msg="  --node-port-algorithm='random'" subsys=daemon
level=info msg="  --node-port-bind-protection='true'" subsys=daemon
level=info msg="  --node-port-mode='snat'" subsys=daemon
level=info msg="  --node-port-range='30000,32767'" subsys=daemon
level=info msg="  --nodes-gc-interval='5m0s'" subsys=daemon
level=info msg="  --operator-api-serve-addr='127.0.0.1:9234'" subsys=daemon
level=info msg="  --policy-audit-mode='false'" subsys=daemon
level=info msg="  --policy-queue-size='100'" subsys=daemon
level=info msg="  --policy-trigger-interval='1s'" subsys=daemon
level=info msg="  --pprof='false'" subsys=daemon
level=info msg="  --pprof-address='localhost'" subsys=daemon
level=info msg="  --pprof-port='6060'" subsys=daemon
level=info msg="  --preallocate-bpf-maps='false'" subsys=daemon
level=info msg="  --prepend-iptables-chains='true'" subsys=daemon
level=info msg="  --procfs='/host/proc'" subsys=daemon
level=info msg="  --prometheus-serve-addr=':9962'" subsys=daemon
level=info msg="  --proxy-connect-timeout='1'" subsys=daemon
level=info msg="  --proxy-gid='1337'" subsys=daemon
level=info msg="  --proxy-max-connection-duration-seconds='0'" subsys=daemon
level=info msg="  --proxy-max-requests-per-connection='0'" subsys=daemon
level=info msg="  --proxy-prometheus-port='0'" subsys=daemon
level=info msg="  --read-cni-conf=''" subsys=daemon
level=info msg="  --remove-cilium-node-taints='true'" subsys=daemon
level=info msg="  --restore='true'" subsys=daemon
level=info msg="  --route-metric='0'" subsys=daemon
level=info msg="  --set-cilium-is-up-condition='true'" subsys=daemon
level=info msg="  --sidecar-istio-proxy-image='cilium/istio_proxy'" subsys=daemon
level=info msg="  --single-cluster-route='false'" subsys=daemon
level=info msg="  --skip-cnp-status-startup-clean='false'" subsys=daemon
level=info msg="  --socket-path='/var/run/cilium/cilium.sock'" subsys=daemon
level=info msg="  --sockops-enable='false'" subsys=daemon
level=info msg="  --srv6-encap-mode='reduced'" subsys=daemon
level=info msg="  --state-dir='/var/run/cilium'" subsys=daemon
level=info msg="  --synchronize-k8s-nodes='true'" subsys=daemon
level=info msg="  --tofqdns-dns-reject-response-code='refused'" subsys=daemon
level=info msg="  --tofqdns-enable-dns-compression='true'" subsys=daemon
level=info msg="  --tofqdns-endpoint-max-ip-per-hostname='50'" subsys=daemon
level=info msg="  --tofqdns-idle-connection-grace-period='0s'" subsys=daemon
level=info msg="  --tofqdns-max-deferred-connection-deletes='10000'" subsys=daemon
level=info msg="  --tofqdns-min-ttl='3600'" subsys=daemon
level=info msg="  --tofqdns-pre-cache=''" subsys=daemon
level=info msg="  --tofqdns-proxy-port='0'" subsys=daemon
level=info msg="  --tofqdns-proxy-response-max-delay='100ms'" subsys=daemon
level=info msg="  --trace-payloadlen='128'" subsys=daemon
level=info msg="  --trace-sock='true'" subsys=daemon
level=info msg="  --tunnel='vxlan'" subsys=daemon
level=info msg="  --tunnel-port='0'" subsys=daemon
level=info msg="  --unmanaged-pod-watcher-interval='15'" subsys=daemon
level=info msg="  --version='false'" subsys=daemon
level=info msg="  --vlan-bpf-bypass=''" subsys=daemon
level=info msg="  --vtep-cidr=''" subsys=daemon
level=info msg="  --vtep-endpoint=''" subsys=daemon
level=info msg="  --vtep-mac=''" subsys=daemon
level=info msg="  --vtep-mask=''" subsys=daemon
level=info msg="  --write-cni-conf-when-ready=''" subsys=daemon
level=info msg="     _ _ _" subsys=daemon
level=info msg=" ___|_| |_|_ _ _____" subsys=daemon
level=info msg="|  _| | | | | |     |" subsys=daemon
level=info msg="|___|_|_|_|___|_|_|_|" subsys=daemon
level=info msg="Cilium 1.13.0 c9723a8d 2023-02-15T14:18:31+01:00 go version go1.19.6 linux/amd64" subsys=daemon
level=info msg="cilium-envoy  version: d59b7f950565d4ccad5c9e30f725740271b7f1c2/1.22.7/Distribution/RELEASE/BoringSSL" subsys=daemon
level=info msg="clang (10.0.0) and kernel (6.1.12) versions: OK!" subsys=linux-datapath
level=info msg="linking environment: OK!" subsys=linux-datapath
level=info msg="Kernel config file not found: if the agent fails to start, check the system requirements at https://docs.cilium.io/en/stable/operations/system_requirements" subsys=probes
level=info msg="Detected mounted BPF filesystem at /sys/fs/bpf" subsys=bpf
level=info msg="Mounted cgroupv2 filesystem at /run/cilium/cgroupv2" subsys=cgroups
level=info msg="Parsing base label prefixes from default label list" subsys=labels-filter
level=info msg="Parsing additional label prefixes from user inputs: []" subsys=labels-filter
level=info msg="Final label prefixes to be used for identity evaluation:" subsys=labels-filter
level=info msg=" - reserved:.*" subsys=labels-filter
level=info msg=" - :io\\.kubernetes\\.pod\\.namespace" subsys=labels-filter
level=info msg=" - :io\\.cilium\\.k8s\\.namespace\\.labels" subsys=labels-filter
level=info msg=" - :app\\.kubernetes\\.io" subsys=labels-filter
level=info msg=" - !:io\\.kubernetes" subsys=labels-filter
level=info msg=" - !:kubernetes\\.io" subsys=labels-filter
level=info msg=" - !:.*beta\\.kubernetes\\.io" subsys=labels-filter
level=info msg=" - !:k8s\\.io" subsys=labels-filter
level=info msg=" - !:pod-template-generation" subsys=labels-filter
level=info msg=" - !:pod-template-hash" subsys=labels-filter
level=info msg=" - !:controller-revision-hash" subsys=labels-filter
level=info msg=" - !:annotation.*" subsys=labels-filter
level=info msg=" - !:etcd_node" subsys=labels-filter
level=info msg="Using autogenerated IPv4 allocation range" subsys=node v4Prefix=10.11.0.0/16
level=info msg=Invoked duration="993.327µs" function="gops.registerGopsHooks (cell.go:39)" subsys=hive
level=info msg=Invoked duration=5.123704ms function="cmd.glob..func2 (daemon_main.go:1643)" subsys=hive
level=info msg="Started gops server" address="127.0.0.1:9890" subsys=gops
level=info msg="Start hook executed" duration="393.209µs" function="gops.registerGopsHooks.func1 (cell.go:44)" subsys=hive
level=info msg="Establishing connection to apiserver" host="https://10.96.0.1:443" subsys=k8s-client
level=info msg="Connected to apiserver" subsys=k8s-client
level=info msg="Start hook executed" duration=6.682159ms function="client.(*compositeClientset).onStart" subsys=hive
level=info msg="Start hook executed" duration=20.239638ms function="cmd.newDatapath.func1 (daemon_main.go:1623)" subsys=hive
level=info msg="Start hook executed" duration="2.62µs" function="*resource.resource[*github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1.Service].Start" subsys=hive
level=info msg="Start hook executed" duration=100.916583ms function="*gobgp.diffStore[*github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1.Service].Start" subsys=hive
level=info msg="Start hook executed" duration="3.8µs" function="*resource.resource[*github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1.CiliumBGPPeeringPolicy].Start" subsys=hive
level=info msg="Start hook executed" duration=760ns function="*resource.resource[*k8s.io/api/core/v1.Node].Start" subsys=hive
level=info msg="Start hook executed" duration=740ns function="*resource.resource[*github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2.CiliumNode].Start" subsys=hive
level=info msg="Start hook executed" duration="15.52µs" function="*agent.ciliumNodeSpecer.Start" subsys=hive
level=info msg="Start hook executed" duration=100.155425ms function="*agent.Controller.Start" subsys=hive
level=info msg="Cilium BGP Control Plane Controller now running..." component=Controller.Run subsys=bgp-control-plane
level=info msg="Cilium BGP Control Plane Controller woken for reconciliation" component=Controller.Run subsys=bgp-control-plane
level=info msg="Using Managed Neighbor Kernel support" subsys=daemon
level=info msg="Registering GoBGP servers for policy with local ASN 64555" component=gobgp.RouterManager.registerBGPServer subsys=bgp-control-plane
level=info msg="Auto-disabling \"enable-node-port\", \"enable-external-ips\", \"bpf-lb-sock\", \"enable-host-port\" features and falling back to \"enable-host-legacy-routing\"" subsys=daemon
level=info msg="Reconciling peers for virtual router with local ASN 64555" component=gobgp.neighborReconciler subsys=bgp-control-plane
level=info msg="Adding peer 192.168.122.1/32 64555 to local ASN 64555" component=gobgp.neighborReconciler subsys=bgp-control-plane
level=info msg="Add a peer configuration" Key=192.168.122.1 Topic=Peer asn=64555 component=gobgp.BgpServerInstance subsys=bgp-control-plane
level=info msg="Done reconciling peers for virtual router with local ASN 64555" component=gobgp.neighborReconciler subsys=bgp-control-plane
level=info msg="Inheriting MTU from external network interface" device=cilium_host ipAddr=10.0.0.11 mtu=1500 subsys=mtu
level=info msg="Successfully registered GoBGP servers for policy with local ASN 64555" component=gobgp.RouterManager.registerBGPServer subsys=bgp-control-plane
level=info msg="type:STATE peer:{conf:{local_asn:64555 neighbor_address:\"192.168.122.1\" peer_asn:64555} state:{local_asn:64555 neighbor_address:\"192.168.122.1\" peer_asn:64555 session_state:IDLE router_id:\"<nil>\"} transport:{local_address:\"<nil>\"}}"
level=info msg="type:STATE peer:{conf:{local_asn:64555 neighbor_address:\"192.168.122.1\" peer_asn:64555} state:{local_asn:64555 neighbor_address:\"192.168.122.1\" peer_asn:64555 session_state:ACTIVE router_id:\"<nil>\"} transport:{local_address:\"<nil>\"}}"
level=info msg="Envoy: Starting xDS gRPC server listening on /var/run/cilium/xds.sock" subsys=envoy-manager
level=info msg="Restored 4 node IDs from the BPF map" subsys=linux-datapath
level=info msg="Restored backends from maps" failedBackends=0 restoredBackends=6 subsys=service
level=info msg="Restored services from maps" failedServices=0 restoredServices=4 subsys=service
level=info msg="Reading old endpoints..." subsys=daemon
level=info msg="Reusing previous DNS proxy port: 44335" subsys=daemon
level=info msg="Waiting until all Cilium CRDs are available" subsys=k8s
level=info msg="All Cilium CRDs have been found and are available" subsys=k8s
level=info msg="Creating or updating CiliumNode resource" node=test-cluster-worker subsys=nodediscovery
level=info msg="Cilium BGP Control Plane Controller woken for reconciliation" component=Controller.Run subsys=bgp-control-plane
level=info msg="Done reconciling peers for virtual router with local ASN 64555" component=gobgp.neighborReconciler subsys=bgp-control-plane
level=info msg="Retrieved node information from cilium node" nodeName=test-cluster-worker subsys=k8s
level=info msg="Received own node information from API server" ipAddr.ipv4=172.18.0.4 ipAddr.ipv6="<nil>" k8sNodeIP=172.18.0.4 labels="map[beta.kubernetes.io/arch:amd64 beta.kubernetes.io/os:linux kubernetes.io/arch:amd64 kubernetes.io/hostname:test-cluster-worker kubernetes.io/os:linux name:node1]" nodeName=test-cluster-worker subsys=k8s v4Prefix=10.0.0.0/24 v6Prefix="<nil>"
level=info msg="k8s mode: Allowing localhost to reach local endpoints" subsys=daemon
level=info msg="Detected devices" devices="[]" subsys=linux-datapath
level=info msg="Enabling k8s event listener" subsys=k8s-watcher
level=info msg="Removing stale endpoint interfaces" subsys=daemon
level=info msg="Waiting until local node addressing before starting watchers depending on it" subsys=k8s-watcher
level=info msg="Skipping kvstore configuration" subsys=daemon
level=info msg="Restored router address from node_config" file=/var/run/cilium/state/globals/node_config.h ipv4=10.0.0.11 ipv6="<nil>" subsys=node
level=info msg="Initializing node addressing" subsys=daemon
level=info msg="Initializing cluster-pool IPAM" subsys=ipam v4Prefix=10.0.0.0/24 v6Prefix="<nil>"
level=info msg="Restoring endpoints..." subsys=daemon
level=info msg="Endpoints restored" failed=0 restored=1 subsys=daemon
level=info msg="Addressing information:" subsys=daemon
level=info msg="  Cluster-Name: default" subsys=daemon
level=info msg="  Cluster-ID: 0" subsys=daemon
level=info msg="  Local node-name: test-cluster-worker" subsys=daemon
level=info msg="  Node-IPv6: <nil>" subsys=daemon
level=info msg="  External-Node IPv4: 172.18.0.4" subsys=daemon
level=info msg="  Internal-Node IPv4: 10.0.0.11" subsys=daemon
level=info msg="  IPv4 allocation prefix: 10.0.0.0/24" subsys=daemon
level=info msg="  Loopback IPv4: 169.254.42.1" subsys=daemon
level=info msg="  Local IPv4 addresses:" subsys=daemon
level=info msg="  - 10.0.0.11" subsys=daemon
level=info msg="  - 172.18.0.4" subsys=daemon
level=info msg="Adding local node to cluster" node="{test-cluster-worker default [{InternalIP 172.18.0.4} {CiliumInternalIP 10.0.0.11}] 10.0.0.0/24 [] <nil> [] 10.0.0.226 <nil> <nil> <nil> 0 local 0 map[beta.kubernetes.io/arch:amd64 beta.kubernetes.io/os:linux kubernetes.io/arch:amd64 kubernetes.io/hostname:test-cluster-worker kubernetes.io/os:linux name:node1] map[] 1 }" subsys=nodediscovery
level=info msg="Creating or updating CiliumNode resource" node=test-cluster-worker subsys=nodediscovery
level=info msg="Waiting until all pre-existing resources have been received" subsys=k8s-watcher
level=info msg="Initializing identity allocator" subsys=identity-cache
level=info msg="Allocating identities between range" cluster-id=0 max=65535 min=256 subsys=identity-cache
level=info msg="Cluster-ID is not specified, skipping ClusterMesh initialization" subsys=daemon
level=info msg="Cilium BGP Control Plane Controller woken for reconciliation" component=Controller.Run subsys=bgp-control-plane
level=info msg="Virtual router with ASN 64555 router ID has changed from 10.0.0.11 to 172.18.0.4" component=gobgp.preflightReconciler subsys=bgp-control-plane
level=info msg="Recreating virtual router with ASN 64555 for changes to take effect" component=gobgp.preflightReconciler subsys=bgp-control-plane
level=info msg="Delete a peer configuration" Key=192.168.122.1 Topic=Peer asn=64555 component=gobgp.BgpServerInstance subsys=bgp-control-plane
level=info msg="Reconciling peers for virtual router with local ASN 64555" component=gobgp.neighborReconciler subsys=bgp-control-plane
level=info msg="Adding peer 192.168.122.1/32 64555 to local ASN 64555" component=gobgp.neighborReconciler subsys=bgp-control-plane
level=info msg="Add a peer configuration" Key=192.168.122.1 Topic=Peer asn=64555 component=gobgp.BgpServerInstance subsys=bgp-control-plane
level=info msg="Done reconciling peers for virtual router with local ASN 64555" component=gobgp.neighborReconciler subsys=bgp-control-plane
level=info msg="type:STATE peer:{conf:{local_asn:64555 neighbor_address:\"192.168.122.1\" peer_asn:64555} state:{local_asn:64555 neighbor_address:\"192.168.122.1\" peer_asn:64555 session_state:IDLE router_id:\"<nil>\"} transport:{local_address:\"<nil>\"}}"
level=info msg="type:STATE peer:{conf:{local_asn:64555 neighbor_address:\"192.168.122.1\" peer_asn:64555} state:{local_asn:64555 neighbor_address:\"192.168.122.1\" peer_asn:64555 session_state:ACTIVE router_id:\"<nil>\"} transport:{local_address:\"<nil>\"}}"
level=info msg="Setting up BPF datapath" bpfClockSource=ktime bpfInsnSet=v3 subsys=datapath-loader
level=info msg="Setting sysctl" subsys=sysctl sysParamName=net.core.bpf_jit_enable sysParamValue=1
level=warning msg="Unable to ensure that BPF JIT compilation is enabled. This can be ignored when Cilium is running inside non-host network namespace (e.g. with kind or minikube)" error="could not open the sysctl file /host/proc/sys/net/core/bpf_jit_enable: open /host/proc/sys/net/core/bpf_jit_enable: no such file or directory" subsys=sysctl sysParamName=net.core.bpf_jit_enable sysParamValue=1
level=info msg="Setting sysctl" subsys=sysctl sysParamName=net.ipv4.conf.all.rp_filter sysParamValue=0
level=info msg="Setting sysctl" subsys=sysctl sysParamName=net.ipv4.fib_multipath_use_neigh sysParamValue=1
level=info msg="Setting sysctl" subsys=sysctl sysParamName=kernel.unprivileged_bpf_disabled sysParamValue=1
level=info msg="Setting sysctl" subsys=sysctl sysParamName=kernel.timer_migration sysParamValue=0
level=info msg="Setting sysctl" subsys=sysctl sysParamName=net.ipv4.conf.cilium_host.forwarding sysParamValue=1
level=info msg="Setting sysctl" subsys=sysctl sysParamName=net.ipv4.conf.cilium_host.rp_filter sysParamValue=0
level=info msg="Setting sysctl" subsys=sysctl sysParamName=net.ipv4.conf.cilium_host.accept_local sysParamValue=1
level=info msg="Setting sysctl" subsys=sysctl sysParamName=net.ipv4.conf.cilium_host.send_redirects sysParamValue=0
level=info msg="Setting sysctl" subsys=sysctl sysParamName=net.ipv4.conf.cilium_net.forwarding sysParamValue=1
level=info msg="Setting sysctl" subsys=sysctl sysParamName=net.ipv4.conf.cilium_net.rp_filter sysParamValue=0
level=info msg="Setting sysctl" subsys=sysctl sysParamName=net.ipv4.conf.cilium_net.accept_local sysParamValue=1
level=info msg="Setting sysctl" subsys=sysctl sysParamName=net.ipv4.conf.cilium_net.send_redirects sysParamValue=0
level=info msg="regenerating all endpoints" reason="one or more identities created or deleted" subsys=endpoint-manager
level=info msg="Iptables rules installed" subsys=iptables
level=info msg="Adding new proxy port rules for cilium-dns-egress:44335" id=cilium-dns-egress subsys=proxy
level=info msg="Iptables proxy rules installed" subsys=iptables
level=info msg="Beginning to read perf buffer" startTime="2023-03-10 15:08:30.339312343 +0000 UTC m=+2.671674828" subsys=monitor-agent
level=info msg="Serving cilium node monitor v1.2 API at unix:///var/run/cilium/monitor1_2.sock" subsys=monitor-agent
level=info msg="Start hook executed" duration=2.150054376s function="cmd.newDaemonPromise.func1 (daemon_main.go:1676)" subsys=hive
level=info msg="Initializing daemon" subsys=daemon
level=info msg="Validating configured node address ranges" subsys=daemon
level=info msg="Starting connection tracking garbage collector" subsys=daemon
level=info msg="Starting IP identity watcher" subsys=ipcache
level=info msg="Initial scan of connection tracking completed" subsys=ct-gc
level=info msg="Regenerating restored endpoints" numRestored=1 subsys=daemon
level=info msg="New endpoint" containerID= datapathPolicyRevision=0 desiredPolicyRevision=0 endpointID=1202 identity=1 ipv4= ipv6= k8sPodName=/ subsys=endpoint
level=info msg="Successfully restored endpoint. Scheduling regeneration" endpointID=1202 subsys=daemon
level=info msg="Datapath signal listener running" subsys=signal
level=info msg="Conntrack garbage collector interval recalculated" deleteRatio=8.802995727161305e-05 newInterval=7m30s subsys=map-ct
level=info msg="Removed endpoint" containerID= datapathPolicyRevision=0 desiredPolicyRevision=0 endpointID=3386 identity=4 ipv4=10.0.0.207 ipv6=10.0.0.207 k8sPodName=/ subsys=endpoint
level=info msg="Launching Cilium health daemon" subsys=daemon
level=info msg="Launching Cilium health endpoint" subsys=daemon
level=info msg="Serving prometheus metrics on :9962" subsys=daemon
level=info msg="Started healthz status API server" address="127.0.0.1:9879" subsys=daemon
level=info msg="Initializing Cilium API" subsys=daemon
level=info msg="Daemon initialization completed" bootstrapTime=2.787485624s subsys=daemon
level=info msg="Configuring Hubble server" eventQueueSize=16384 maxFlows=4095 subsys=hubble
level=info msg="Starting local Hubble server" address="unix:///var/run/cilium/hubble.sock" subsys=hubble
level=info msg="Starting Hubble server" address=":4244" subsys=hubble
level=info msg="Serving cilium API at unix:///var/run/cilium/cilium.sock" subsys=daemon
level=info msg="Compiled new BPF template" BPFCompilationTime=673.734749ms file-path=/var/run/cilium/state/templates/5de6d856d7d468f2347401628fec2ae7b165cf61d961cb7ea05d00ef08046609/bpf_host.o subsys=datapath-loader
level=info msg="Rewrote endpoint BPF program" containerID= datapathPolicyRevision=0 desiredPolicyRevision=1 endpointID=1202 identity=1 ipv4= ipv6= k8sPodName=/ subsys=endpoint
level=info msg="Restored endpoint" endpointID=1202 ipAddr="[ ]" subsys=endpoint
level=info msg="Finished regenerating restored endpoints" regenerated=1 subsys=daemon total=1
level=info msg="New endpoint" containerID= datapathPolicyRevision=0 desiredPolicyRevision=0 endpointID=1811 ipv4= ipv6= k8sPodName=/ subsys=endpoint
level=info msg="Resolving identity labels (blocking)" containerID= datapathPolicyRevision=0 desiredPolicyRevision=0 endpointID=1811 identityLabels="reserved:health" ipv4= ipv6= k8sPodName=/ subsys=endpoint
level=info msg="Identity of endpoint changed" containerID= datapathPolicyRevision=0 desiredPolicyRevision=0 endpointID=1811 identity=4 identityLabels="reserved:health" ipv4= ipv6= k8sPodName=/ oldIdentity="no identity" subsys=endpoint
level=info msg="Compiled new BPF template" BPFCompilationTime=2.417421162s file-path=/var/run/cilium/state/templates/82cc0e78b37ae2713f287ca676521f5b6e141952d5a4b7996d3c0576e447300b/bpf_lxc.o subsys=datapath-loader
level=info msg="Rewrote endpoint BPF program" containerID= datapathPolicyRevision=0 desiredPolicyRevision=1 endpointID=1811 identity=4 ipv4= ipv6= k8sPodName=/ subsys=endpoint
level=info msg="type:STATE peer:{conf:{local_asn:64555 neighbor_address:\"192.168.122.1\" peer_asn:64555} state:{local_asn:64555 neighbor_address:\"192.168.122.1\" peer_asn:64555 session_state:OPENSENT router_id:\"<nil>\"} transport:{local_address:\"172.18.0.4\" local_port:33277 remote_port:179}}"
level=info msg="Peer Up" Key=192.168.122.1 State=BGP_FSM_OPENCONFIRM Topic=Peer asn=64555 component=gobgp.BgpServerInstance subsys=bgp-control-plane
level=info msg="type:STATE peer:{conf:{local_asn:64555 neighbor_address:\"192.168.122.1\" peer_asn:64555} state:{local_asn:64555 neighbor_address:\"192.168.122.1\" peer_asn:64555 session_state:OPENCONFIRM router_id:\"172.17.0.1\"} transport:{local_address:\"172.18.0.4\" local_port:33277 remote_port:179}}"
level=info msg="type:STATE peer:{conf:{local_asn:64555 neighbor_address:\"192.168.122.1\" peer_asn:64555} state:{local_asn:64555 neighbor_address:\"192.168.122.1\" peer_asn:64555 session_state:ESTABLISHED router_id:\"172.17.0.1\"} transport:{local_address:\"172.18.0.4\" local_port:33277 remote_port:179}}"
level=info msg="Serving cilium health API at unix:///var/run/cilium/health.sock" subsys=health-server
```

**Comment by dylandreimerink** [comment]
@NikAleksandrov thank you, I am not able to reproduce the issue, will update this issue once I have found the root cause.

**Comment by dylandreimerink** [comment]
I found the cause of the bug. Explaining it is a bit more difficult. Turns out there is an edge-case in the reconciliation logic which is triggered when we re-create the BGP server. This normally doesn't happen that often if it was not for some odd behavior in IPAM which causes the default route-id to update during startup.

This results in the following sequence of events:
1. BGP Subsystem starts, reads the current CiliumNode object + PeeringPolicy
2. The PreflightReconciler makes a new GoBGP server and the PeerReconciler a new connection
3. The PodCIDRReconciler makes the announcement, writes it to the GoBGP server and to a slice on our side
4. Some IPAM logic removes the CiliumInternalIP from the CiliumNode and re-adds it (exact cause still unknown)
5. The PreflightReconciler sees the change in route-id and re-creates the GoBGP server, thus clearing all tables. But the slice of announcements on our side is not cleared.
5. The PodCIDRReconciler checks if the PodCIDR is announced, but does so against our slice and never goes out to GoBGP to check if the announcement exists. It concludes that there is nothing to be done.

So we end up in a situation where we are missing routes. This indeed effects both PodCIDR and Service announcements since both use such a shadow map. It also explains why removing the peering policy and re-adding it works since this also clears the shadow data.

Clearing this shadow data turns out to be a 2 liner, PR #24304

However, we should re-consider the current design where we maintain the shadow data in this way, we should remove it or sync it against the GoBGP server table so we maintain a clear source of truth

**Comment by Ramblurr** [comment]
@dylandreimerink I think I am being affected by this bug.. I see that the PR has been merged but not released yet. Is there a workaround? Or is it possible to run the `master` version with helm (not in prod, but to test this fix)?

**Comment by dylandreimerink** [comment]
We are backporting the fix to v1.12.9 and v1.13.2. Until they are released, a workaround I can think of would be to manually set the `router-id` with annotations https://docs.cilium.io/en/latest/network/bgp-control-plane/#router-id-attribute. I think this should work since the bug is triggered by the nodes primary IP changing back and forth just after agent startup. By setting the router-id with the annotation instead of inferring it from the primary IP we should avoid this trigger, however I have not tested this yet.

**Comment by kakkotetsu** [comment]
I tried that workaround and it did not work in my environment.(Cilium 1.13.0 + Kubernetes 1.26.2)

**Comment by dylandreimerink** [comment]
@kakkotetsu I am sorry to hear that. Thank you for trying. Then there is no effective workaround. Re-creating the peering policy resolve it temporarily, until the next agent restart. So that is a very temporary bandage.

In theory, you can manually compile and build cilium, but any locally created image will lack the SBOM and not be signed. Also changing the helm values to get the image to deploy. Unfortunately, we currently have no instructions on how to do all of this.

Given the normal patch release cadence I expect the official images to be released in about 2-3 weeks.

**Comment by Ramblurr** [comment]
Thanks. We've rolled back to the old [`bgp`](https://docs.cilium.io/en/v1.13/network/bgp/) impl in the mean time.
